lib/server.js

```diff
@@ -171,23 +172,41 @@ Server.prototype._start = function (config, launcher, preprocess, fileList,
       self._injector.invoke(watcher.watch)
     }
 
-    webServer.listen(config.port, config.listenAddress, function () {
-      self.log.info('Karma v%s server started at %s//%s:%s%s', constant.VERSION,
-        config.protocol, config.listenAddress, config.port, config.urlRoot)
+    var startWebServer = function () {
+      webServer.listen(config.port, config.listenAddress, function () {
+        self.log.info('Karma v%s server started at %s//%s:%s%s', constant.VERSION,
+          config.protocol, config.listenAddress, config.port, config.urlRoot)
 
-      self.emit('listening', config.port)
-      if (config.browsers && config.browsers.length) {
-        self._injector.invoke(launcher.launch, launcher).forEach(function (browserLauncher) {
-          singleRunDoneBrowsers[browserLauncher.id] = false
-        })
-      }
-      var noLoadErrors = self.loadErrors.length
-      if (noLoadErrors > 0) {
-        self.log.error('Found %d load error%s', noLoadErrors, noLoadErrors === 1 ? '' : 's')
-        process.exitCode = 1
-        process.kill(process.pid, 'SIGINT')
-      }
-    })
+        self.emit('listening', config.port)
+        if (config.browsers && config.browsers.length) {
+          self._injector.invoke(launcher.launch, launcher).forEach(function (browserLauncher) {
+            singleRunDoneBrowsers[browserLauncher.id] = false
+          })
+        }
+        var noLoadErrors = self.loadErrors.length
+        if (noLoadErrors > 0) {
+          self.log.error('Found %d load error%s', noLoadErrors, noLoadErrors === 1 ? '' : 's')
+          process.exitCode = 1
+          process.kill(process.pid, 'SIGINT')
+        }
+      })
+    }
+
+    const karmaJsPath = path.join(__dirname, '/../static/karma.js')
+    const contextJsPath = path.join(__dirname, '/../static/context.js')
+    // Check if the static files haven't been compiled
+    if (!(fs.existsSync(karmaJsPath) && fs.existsSync(contextJsPath))) {
```

Comment:
please use the async versions

Reply:
`fs.exists` is deprecated, `fs.existsSync` is not. And it seems like it would be an unnecessary complication to use `fs.stat` on two separate files and make callbacks that `&&` the results together.